.github/workflows/codeql.yml

```diff
@@ -0,0 +1,71 @@
+name: "CodeQL Advanced"
+
+on:
+  push:
+  pull_request:
+    paths-ignore:
+    - '**.md'
+  schedule:
+    - cron: '43 2 * * 5'
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ${{ (matrix.language == 'c-cpp' && 'windows-latest') || 'ubuntu-latest' }}
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # required to fetch internal or private CodeQL packs
+      packages: read
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - language: actions
+          build-mode: none
+        - language: c-cpp
+          build-mode: manual
+        - language: javascript-typescript
```

Comment:
I feel like we don't really have an js/python in the repo. What are your thoughts on just limiting it to the src/ directory  + evaluating c-cpp (maybe some obj c)? Open to opinions on this.

<img width="309" alt="image" src="https://github.com/user-attachments/assets/5cf8a1cb-0277-4fcb-97ef-0b0065ec2618" />

Reply:
Makes sense, I've limited it to just c-cpp under the src/ directory. It doesn't support Objective-C (or Lua) unfortunately - https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/#id3 - and was scanning one JavaScript and three Python files.

Not sure if you're able to see things under https://github.com/itgmania/itgmania/security/code-scanning but on my fork about half of the notifications raised were under the extern/ directory

Reply:
Yeah I think just c-cpp seems good:

The link shows me:
![image](https://github.com/user-attachments/assets/d590bba1-b2b9-45fa-bdda-34e36b5e5398)

BUT I do recall seeing some stuff way earlier and it was a ton of extern as well.

Reply:
Looks like we won't be able to filter to the src directory unfortunately - https://github.com/itgmania/itgmania/actions/runs/14271424189/job/40005341525#step:3:720